plugin/pkg/scheduler/scheduler.go

```diff
@@ -219,9 +219,6 @@ func (sched *Scheduler) scheduleOne() {
 		// If binding succeeded then PodScheduled condition will be updated in apiserver so that
 		// it's atomic with setting host.
 		err := sched.config.Binder.Bind(b)
-		if err := sched.config.SchedulerCache.FinishBinding(&assumed); err != nil {
-			glog.Errorf("scheduler cache FinishBinding failed: %v", err)
-		}
 		if err != nil {
 			glog.V(1).Infof("Failed to bind pod: %v/%v", pod.Namespace, pod.Name)
 			if err := sched.config.SchedulerCache.ForgetPod(&assumed); err != nil {
```

Comment:
Let me take some time to find a case to trigger the following case: 
https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/scheduler/schedulercache/cache.go#L164

If it's triggered, the assumed pod will not be deleted after this PR.